### ISSUE TITLE
Error when live reload.

### ISSUE THREAD

**Issue by ThunderBirdsX3** [ionic-team/ionic#13015]
**Ionic version:**  (check one with "x")
(For Ionic 1.x issues, please use https://github.com/ionic-team/ionic-v1)
[ ] **2.x**
[x] **3.x**
[ ] **4.x**

**I'm submitting a ...**  (check one with "x")
[x] bug report
[ ] feature request

**Current behavior:**
Always got an error when live reload after save the line of code. Seem like it not go back to start page.

That like if you set `rootPage:any = 'TabsPage';`, In first time when run `ionic cordova emulate ios -l -c --target=iPhone-6`, It will start with `TabsPage`.
 But if i go to `SomePage` with `this.app.getRootNavs()[0].setRoot('SomePage')` then `this.navCtrl.push('OtherPage')` from `SomePage`, And save new line of code in `OtherPage` it will broken. I must to run new command for this problem.

**Expected behavior:**
Save new line of code and start with `TabsPage`.

**Ionic info:** (run `ionic info` from a terminal/cmd prompt and paste output below):

```
cli packages: (/usr/local/lib/node_modules)

    @ionic/cli-utils  : 1.12.0
    ionic (Ionic CLI) : 3.12.0

global packages:

    cordova (Cordova CLI) : 7.0.1 

local packages:

    @ionic/app-scripts : 2.1.4
    Cordova Platforms  : ios 4.5.0
    Ionic Framework    : ionic-angular 3.7.0

System:

    ios-deploy : 1.9.2 
    ios-sim    : 6.1.2 
    Node       : v6.11.0
    npm        : 5.4.2 
    OS         : macOS Sierra
    Xcode      : Xcode 9.0 Build version 9A235 

Misc:

    backend : legacy
```

**Comment by kensodemann** [comment]
Hello! Thank you for opening an issue with us!

Are you using lazy loading or deep linking at all? If so, then this is expected behavior. The reload just reloads whatever URL you are on.

Thank you for using Ionic

**Comment by ThunderBirdsX3** [comment]
@kensodemann I used lazy loading.

Order when call page is.
1. `rootPage:any = 'LoadingPage';` for check information and check login.
2. `this.navCtrl.setRoot('TabsPage').catch(() => {});` After check info go to `TabsPage`.
3. `this.app.getRootNavs()[0].push('SubjectPage').catch(()=>{});` In `AboutPage`  in Tabs.
4. `this.navCtrl.push('SubjectMessagePage', { subject: subject});` from `SubjectPage`.
5. Update code or provider it used in `SubjectPage`.

In `SubjectPage` it have socket.io, After I send or receive data with API and socket.io it will broken.

Socket.io with this package https://github.com/ThunderBirdsX3/echo

**Comment by kensodemann** [comment]
Would you be able to provide a sample application via GitHub that demonstrates this issue? What you have above is very hard to follow. I will follow this up with a comment on what I think you are describing, but what you posted above is quite hard to follow out of context.

**Comment by kensodemann** [comment]
Here is what I am hearing in your posts:

** Original Issue **
1.  start with `http://localhost:8100` which goes to `http://localhost:8100/#/loading-page`
2. `http://localhost:8100/#/tabs-page/home-page` (or something like that after re-direct from loading)
3. `http://localhost:8100/#/tabs-page/some-page`
4. `http://localhost:8100/#/tabs-page/some-page/other-page`

At this point you change some code and are expecting the reload to take you back to the start, but it is not going to. It is going to take you to `http://localhost:8100/#/tabs-page/some-page/other-page` because that is the way a refresh works. It refreshes the current URL.

** Current Issue **

1. basically the same steps as above
2. the issue isn't that it reloads `other-page` so much as you have an error on `other-page`


At least that is how I am understanding this. Like is said, what you have is very hard to follow out of context, so describing it within the context of a sample application may be helpful.

**Comment by kensodemann** [comment]
Closing due to inactivity. If you can provide a sample application via GitHub that we can use to troubleshoot this issue I will be happy to reopen it.
